### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by midasdev711**
### Problem

- In the ultrawide monitor, it's kind of hard to read all the numbers as they are really separated and all over the place.
- Top left corner logo is blurry.
- Nav bar buttons don't have action when hovering.
- It makes no sense when the mouse is on top of #2 row.

### Solution

- Set the max-width of the content
- Replace the logo
- Change color of nav bar link when hovering
- Remove hover effect in all tables

### Result

All issues are fixed.

image

### CODE REVIEW

File: web-ui/src/environments/environment.ts [hunk 1]
```diff
@@ -6,7 +6,8 @@
 export const environment = {
   production: false,
   api: {
-    url: 'http://localhost:9000'
+    url: 'https://xsnexplorer.io/api',
```
Comment (adinael) [inline web-ui/src/environments/environment.ts:9]: this line should not be changed.

File: web-ui/angular.json [hunk 2]
```diff
@@ -21,9 +21,11 @@
               "src/favicon.ico"
             ],
             "styles": [
-              "src/styles.css"
+              "src/styles.css",
+              "./node_modules/ngx-ui-switch/ui-switch.component.css"
             ],
-            "scripts": []
+            "scripts": ["node_modules/jquery/dist/jquery.min.js",
```
Comment (AlexITC) [inline web-ui/angular.json]: Mixing jquery with angular is usually a bad idea, what approaches have you considered that led you to still require jquery?

File: web-ui/angular.json [hunk 3]
```diff
@@ -21,9 +21,11 @@
...
+              "node_modules/bootstrap/dist/js/bootstrap.js"]
```
Comment (AlexITC) [inline web-ui/angular.json]: I think the explorer already uses https://valor-software.com/ngx-bootstrap/#/, what's the need behind bringing bootstrap separately?

File: web-ui/angular.json [hunk 4]
```diff
@@ -21,9 +21,11 @@
               "src/favicon.ico"
             ],
             "styles": [
-              "src/styles.css"
+              "src/styles.css",
+              "./node_modules/ngx-ui-switch/ui-switch.component.css"
```
Comment (AlexITC) [inline web-ui/angular.json:25]: Is this component ever used?

File: web-ui/package.json [hunk 5]
```diff
@@ -4,14 +4,17 @@
   "license": "MIT",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "serve": "ng serve",
+    "start": "node server.js",
     "build": "ng build",
-    "test": "ng test",
+    "test": "ng test --no-watch --no-progress --browsers=ChromeHeadlessCI",
```
Comment (AlexITC) [inline web-ui/package.json]: Add a new command instead

File: web-ui/package.json [hunk 6]
```diff
@@ -4,14 +4,17 @@
...
     "lint": "ng lint",
-    "e2e": "ng e2e"
+    "e2e": "ng e2e",
+    "prod": "ng build --prod"
```
Comment (AlexITC) [inline web-ui/package.json]: Rename this to `buiild-prod` instead

File: web-ui/package.json [hunk 7]
```diff
@@ -4,14 +4,17 @@
...
   },
   "private": true,
   "dependencies": {
+    "@angular-devkit/build-angular": "^0.10.7",
```
Comment (AlexITC) [inline web-ui/package.json:17]: While this is nice, you will likely have a hard time getting the tests to pass, what has forced you to upgrade?

File: web-ui/package.json [hunk 8]
```diff
@@ -22,23 +25,32 @@
     "@angular/platform-browser-dynamic": "^7.2.1",
     "@angular/router": "^7.2.1",
     "@ngx-translate/core": "^10.0.2",
+    "@syncfusion/ej2-angular-inputs": "^18.4.47",
```
Comment (AlexITC) [inline web-ui/package.json:28]: How is this being used?

File: web-ui/package.json [hunk 9]
```diff
@@ -22,23 +25,32 @@
...
     "babel-polyfill": "^6.26.0",
     "bootstrap": "^3.4.1",
+    "connect-history-api-fallback": "^1.6.0",
```
Comment (AlexITC) [inline web-ui/package.json:31]: How is this being used?

File: web-ui/package.json [hunk 10]
```diff
@@ -22,23 +25,32 @@
...
     "core-js": "^2.5.4",
+    "express": "^4.16.4",
```
Comment (AlexITC) [inline web-ui/package.json:33]: This is a backend dependency being added to a frontend app, I'm interested to know if this is really required and why

File: web-ui/package.json [hunk 11]
```diff
@@ -22,23 +25,32 @@
...
+    "jquery": "^3.5.1",
```
Comment (AlexITC) [inline web-ui/package.json]: As pointed out in another commnet, jquery is usually not recommended in angular projects, what prevent us to remove it?

File: web-ui/package.json [hunk 12]
```diff
@@ -22,23 +25,32 @@
...
+    "lottie-web": "^5.7.6",
```
Comment (AlexITC) [inline web-ui/package.json:34]: Can you give me some examples where this library is used?

File: web-ui/server.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const path = require('path');
+const history = require('connect-history-api-fallback');
+
+const app = express();
```
Comment (AlexITC) [inline web-ui/server.js:5]: Isn't this already handled by angular?

File: web-ui/tsconfig.json [hunk 14]
```diff
@@ -7,6 +7,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
+    "skipLibCheck": true,
```
Comment (AlexITC) [inline web-ui/tsconfig.json:10]: From the official [docs](https://www.typescriptlang.org/tsconfig#skipLibCheck), its preferred to make sure that the same library has a single version in our code, try that instead of adding this config

File: web-ui/src/styles.css [hunk 15]
```diff
@@ -35,3 +49,414 @@ body {
...
+  height: 40px;
+  background: #222534 0% 0% no-repeat padding-box;
+  border-radius: 8px;
+  color: #989DCC;
+  font-size: 14px;
+  padding-left: 19px;
+  padding-right: 16px;
+  border: none;
+  appearance: none;
+  background: #222534 url("data:image/svg+xml;utf8,<svg viewBox='0 0 140 140' width='24' height='24' xmlns='http://www.w3.org/2000/svg' ><g><path d='m121.3,34.6c-1.6-1.6-4.2-1.6-5.8,0l-51,51.1-51.1-51.1c-1.6-1.6-4.2-1.6-5.8,0-1.6,1.6-1.6,4.2 0,5.8l53.9,53.9c0.8,0.8 1.8,1.2 2.9,1.2 1,0 2.1-0.4 2.9-1.2l53.9-53.9c1.7-1.6 1.7-4.2 0.1-5.8z' fill='%23989DCC'/></g></svg>") calc(100% - 10px) 50% no-repeat;
```
Comment (AlexITC) [inline web-ui/src/styles.css:309]: Wouldn't it be better to take this svg to its own file?

File: web-ui/src/styles.css [hunk 16]
```diff
@@ -35,3 +49,414 @@ body {
...
+  .no-left,
+  .no-right {
+    padding-left: unset;
+    padding-right: unset;
+  }
+}
+
+@media (min-width:992px) {
+  .no-left {
+    padding-left: 0 !important;
```
Comment (AlexITC) [inline web-ui/src/styles.css:239]: `!important` is something that is generally avoided unless there is no other way, I see quite a lot of times this is being used, please consider avoiding it

File: web-ui/src/index.html [hunk 17]
```diff
@@ -27,6 +27,10 @@
 
     gtag('config', 'UA-65008315-3');
   </script>
+  <script>window.__theme = 'bs3';</script>
+  <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"></script>
```
Comment (AlexITC) [inline web-ui/src/index.html:32]: Try not mixing two ways for importing scripts, npm is the way we are using, also, npm imports jquery which is imported again.

File: web-ui/src/assets/Updating.json [hunk 18]
```diff
@@ -0,0 +1 @@
...:{"a":0,"k":[49.522,45.788,0],"ix":2},"a":{"a":0,"k":[0,0,0],"ix":1},"s":{"a":1,"k":[{"i":{"x":[0.667,0.667,0.667],"y":[1,1,1]},"o":{"x":[0.333,0.333,0.333],"y":[0,0,0]},"t":0,"s":[20,20,100]},{"t":31.0000012626559,"s":[100,100,100]}],"ix":6}},"ao":0,"shapes":[{"ty":"gr","it":[{"d":1,"ty":"el","s":{"a":0,"k":[91.375,91.375],"ix":2},"p":{"a":0,"k":[0,0],"ix":3},"nm":"Ellipse Path 1","mn":"ADBE Vector Shape - Ellipse","hd":false},{"ty":"st","c":{"a":0,"k":[1,1,1,1],"ix":3},"o":{"a":0,"k":100,"ix":4},"w":{"a":0,"k":0,"ix":5},"lc":1,"lj":1,"ml":4,"bm":0,"nm":"Stroke 1","mn":"ADBE Vector Graphic - Stroke","hd":false},{"ty":"fl","c":{"a":0,"k":[0.00784313725490196,0.8509803921568627,0.8745098039215686,1],"ix":4},"o":{"a":0,"k":100,"ix":5},"r":1,"bm":0,"nm":"Fill 1","mn":"ADBE Vector Graphic - Fill","hd":false},{"ty":"tr","p":{"a":0,"k":[-1.522,2.212],"ix":2},"a":{"a":0,"k":[0,0],"ix":1},"s":{"a":0,"k":[105.072,104.426],"ix":3},"r":{"a":0,"k":0,"ix":6},"o":{"a":0,"k":100,"ix":7},"sk":{"a":0,"k":0,"ix":4},"sa":{"a":0,"k":0,"ix":5},"nm":"Transform"}],"nm":"Ellipse 1","np":3,"cix":2,"bm":0,"ix":1,"mn":"ADBE Vector Group","hd":false}],"ip":0,"op":150.000006109625,"st":0,"bm":0}],"markers":[]}
```
Comment (AlexITC) [inline web-ui/src/assets/Updating.json:1]: What's the purpose of this file? also, make sure to be consistent with the names, this is the only one starting with upper case.

File: web-ui/src/app/app.module.ts [hunk 19]
```diff
@@ -1,22 +1,19 @@
...
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA  } from '@angular/core';
+import { HttpClientModule } from '@angular/common/http';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 
 import { AppRoutingModule } from './app-routing.module';
-import { AlertModule, BsDropdownModule, CollapseModule, TooltipModule, ModalModule } from 'ngx-bootstrap';
+import { AlertModule } from 'ngx-bootstrap/alert';
 import { TabsModule } from 'ngx-bootstrap/tabs';
+import { BsDropdownModule } from 'ngx-bootstrap/dropdown';
+// import { NgxBootstrapSliderModule } from 'ngx-bootstrap-slider';
```
Comment (AlexITC) [inline web-ui/src/app/app.module.ts]: Remove if this is not used

File: web-ui/src/app/app.component.html [hunk 20]
```diff
@@ -1,6 +1,10 @@
...
+<div class="tab-container">
+  <app-navbar [tabs]="tabs"></app-navbar>
+</div>
+<div class="tab-container">
   <router-outlet></router-outlet>
 </div>
-<ng-http-loader [filteredUrlPatterns]="['/blocks$']"></ng-http-loader>
-<app-footer></app-footer>
+<div class="tab-container">
+  <ng-http-loader [filteredUrlPatterns]="['/blocks$']"></ng-http-loader>
```
Comment (AlexITC) [inline web-ui/src/app/app.component.html:8]: I think that `http-loader` doesn't work anymore

File: web-ui/src/app/utils.ts [hunk 21]
```diff
@@ -4,3 +4,42 @@ export function getNumberOfRowsForScreen(height: number): number {
     }
     return Math.min(10 + Math.ceil((height - 550) / 20), 100);
 }
+
+export function truncate(fullStr, frontChars, backChars, separator = null) {
```
Comment (AlexITC) [inline web-ui/src/app/utils.ts:8]: Typescript is useless if you use everything as `any`, make sure to specify types.

File: web-ui/src/app/utils.ts [hunk 22]
```diff
@@ -4,3 +4,42 @@ export function getNumberOfRowsForScreen(height: number): number {
...
+        return fullStr;
+    }
+
+    separator = separator || ' ... ';
+    return fullStr.substr(0, frontChars) +
+        separator +
+        fullStr.substr(fullStr.length - backChars);
+}
+
+export function amAgo(timestamp: number): string {
```
Comment (AlexITC) [inline web-ui/src/app/utils.ts:19]: Use a function name that's clear enough, also, include tests.

If this function was copied from somewhere else, add the source in a comment, also, consider using moment.js which already handles the corner cases properly

File: web-ui/src/app/utils.ts [hunk 23]
```diff
@@ -4,3 +4,42 @@ export function getNumberOfRowsForScreen(height: number): number {
...
+        i++;
+      } else {
+        break;
+      }
+    }
+
+    return diff.toFixed(0) + units[i].unit + ' ago';
+}
+
+export function numberWithCommas(x) {
```
Comment (AlexITC) [inline web-ui/src/app/utils.ts:43]: Typescript is useless if you use everything as `any`, make sure to specify types.

Also, add tests

File: web-ui/src/app/utils.spec.ts [hunk 24]
```diff
@@ -0,0 +1,13 @@
+import {truncate} from './utils';
+
+describe('Utils testing', () => {
+  it('should truncate the string', (() => {
```
Comment (AlexITC) [inline web-ui/src/app/utils.spec.ts:4]: Add better tests

File: web-ui/src/app/pipes/explorer-amount.pipe.ts [hunk 25]
```diff
@@ -4,16 +4,32 @@ import { Pipe, PipeTransform } from '@angular/core';
...
+    let digits = number.split(".");
+    if (digits.length > 1) {
+      width -= digits.pop().length;
+    } else {
+      number += ".";
+    }
 
-  transform(currency: any): string {
+    if ( width > 0 )
+    {
```
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: This is not consistent to how the if-else are being written in the code, we need a code formatter

File: web-ui/src/app/pipes/explorer-amount.pipe.ts [hunk 26]
```diff
@@ -4,16 +4,32 @@ import { Pipe, PipeTransform } from '@angular/core';
...
+      return number + new Array( width + 1 ).join( '0' );
+    }
+    return number + ""; // always return a string
+  }
+
+  transform(currency: any, digit: number = 8, fillZero:Boolean = false): string {
```
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: > fillZero:Boolean

Add space after the colon

File: web-ui/src/app/pipes/explorer-amount.pipe.ts [hunk 27]
```diff
@@ -4,16 +4,32 @@ import { Pipe, PipeTransform } from '@angular/core';
   name: 'explorerAmount'
 })
 export class ExplorerAmountPipe implements PipeTransform {
+  zeroFill( number: string, width: number ): string {
```
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: Its not clear what this does, ideally, add a couple of tests

File: web-ui/src/app/pipes/pipes.module.ts [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+import { RouterModule } from '@angular/router';
+import { TranslateModule } from '@ngx-translate/core';
+import { MomentModule } from 'ngx-moment';
+import { ExplorerAmountPipe } from './explorer-amount.pipe';
+import { ExplorerCurrencyPipe } from './explorer-currency.pipe';
+import { ExplorerDatetimePipe } from './explorer-datetime.pipe';
+
+@NgModule({
+    imports: [
+        CommonModule,
```
Comment (AlexITC) [inline web-ui/src/app/pipes/pipes.module.ts:13]: I don't see how the pipes module could depend on forms/router/etc, make sure to do some clean up in the other modules too

**Review by AlexITC — CHANGES_REQUESTED** [review]
I left some preliminary comments, overall, please take the comments for anything else that I haven't reviewed, these are some patterns I found:
- Its not clear that every dependency is necessary, from the npm ones to the modules depending on other modules.
- The code style is not consistent to what is already in the code, sometimes it isn't even consistent with the new code being written.
- Some function/variable names aren't clear.
- Lack of testing for important functions like money transformation, while I don't expect extensive testing, at least the core functionality can have a decent coverage.
- Usage of `any` when it doesn't seem necessary, Typescript is useless in this case.


Thanks.

File: web-ui/src/app/pipes/explorer-amount.pipe.ts (hunk 27 above)
Comment (midasdev711) [reply web-ui/src/app/pipes/explorer-amount.pipe.ts]: That's for this case.

{{ 8456.789 | explorerAmount }}  => 8,456.789
{{ 8456.789 | explorerAmount:4:true }}  => 8,456.7890

File: web-ui/angular.json (hunk 4 above)
Comment (midasdev711) [reply web-ui/angular.json:25]: Yes, in dex host calculator

File: web-ui/angular.json (hunk 3 above)
Comment (midasdev711) [reply web-ui/angular.json]: I removed them.

File: web-ui/angular.json (hunk 2 above)
Comment (midasdev711) [reply web-ui/angular.json]: Removed it.

File: web-ui/package.json [hunk 29]
```diff
@@ -4,14 +4,17 @@
   "license": "MIT",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "serve": "ng serve",
+    "start": "node server.js",
```
Comment (AlexITC) [inline web-ui/package.json:8]: Please explain what's the usefulness to include express because angular already has an embedded server for dev purposes, if this is not necessary, remove it

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts [hunk 30]
```diff
@@ -0,0 +1,151 @@
...
+})
+export class AddressDetailsComponent implements OnInit {
+
+  address: Balance;
+  addressString: string;
+  addressLabel = addressLabels;
+  tposContracts: Array<TposContract>;
+  selectedTpos: number;
+  isLoading: boolean;
+  loadingType = 2;
```
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts:28]: Use a enum instead of int values, or at least, create constants representing the possible statuses instead of using magic numbers

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts [hunk 31]
```diff
@@ -0,0 +1,151 @@
...
+  addressLoaded: boolean;
+  interval = null;
```
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts]: What's this for? you forgot to define the type

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts [hunk 32]
```diff
@@ -0,0 +1,151 @@
...
+  styleUrls: ['./address-details.component.css']
+})
+export class AddressDetailsComponent implements OnInit {
+
+  address: Balance;
+  addressString: string;
+  addressLabel = addressLabels;
+  tposContracts: Array<TposContract>;
+  selectedTpos: number;
+  isLoading: boolean;
```
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts:27]: you likely need a `isLoading` per component loaded asynchronously, otherwise, there will be conflicts while loading transactions and tpos contracts

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts [hunk 33]
```diff
@@ -0,0 +1,151 @@
...
+    private router: Router,
+    private addressesService: AddressesService,
+    private errorService: ErrorService) { 
+      this.selectedTpos = 0;
+      this.addressString = null;
+    }
+
+  ngOnInit() {
+    const height = this.getScreenSize();
+    // this.limit = getNumberOfRowsForScreen(height);
```
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts]: Remove the unused code instead of commenting it, unless there is a reason to keep, in which case, add the necessary comment

File: web-ui/src/app/components/blocks/block-list/block-list.component.ts [hunk 34]
```diff
@@ -0,0 +1,75 @@
...
+export class BlockListComponent implements OnInit {
+
+    ticker: ServerStats = new ServerStats();
+    prices: Prices = new Prices();
+    stats: ServerStats = new ServerStats();
+    blocks: Block[] = [];
+    limit = 20;
+    lastSeenHash: string;
+    isLoading: Boolean = false;
+    loadingType = 2;
```
Comment (AlexITC) [inline web-ui/src/app/components/blocks/block-list/block-list.component.ts:22]: Same comment about magic numbers

File: web-ui/src/app/components/calculator/calculator.component.ts [hunk 35]
```diff
@@ -0,0 +1,328 @@
...
+  styleUrls: ['./calculator.component.css']
+})
+export class CalculatorComponent implements OnInit, AfterViewChecked {
+
+  transaction: any;
+  prices: Prices = new Prices();
+  stats: NodeStats = new NodeStats();
+  rewardsSummary: RewardsSummary = new RewardsSummary();
+  serverStats: ServerStats = new ServerStats();
+  holdAmount = null;
```
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts]: Missing type

File: web-ui/src/app/components/calculator/calculator.component.ts [hunk 36]
```diff
@@ -0,0 +1,328 @@
...
+
+  stakingWaitTime = "";
+  mnWaitTime = "";
+
+  interval: any;
+
+  // hydra node
+  xsnPriceLog = 1;
+  xsnPrice = 1;
+  xsnPriceInputValue = null;
```
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts]: Make sure to address the missing type specified in most of the changes

File: web-ui/src/app/components/calculator/calculator.component.ts [hunk 37]
```diff
@@ -0,0 +1,328 @@
...
+import { TickerService } from '../../services/ticker.service';
+import { XSNService } from '../../services/xsn.service';
+import { RewardsSummary, NodeStats, Prices } from '../../models/xsn';
+
+@Component({
+  selector: 'app-calculator',
+  templateUrl: './calculator.component.html',
+  styleUrls: ['./calculator.component.css']
+})
+export class CalculatorComponent implements OnInit, AfterViewChecked {
```
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts:13]: This component has quite a lot of functions that could be taken out of the component, make sure to tests those, its pretty important to get the correct data for users

File: .circleci/config.yml [hunk 38]
```diff
@@ -18,7 +18,7 @@ jobs:
           paths:
             - "web-ui/node_modules"
       - run: cd web-ui && npm run test -- --no-watch --no-progress --browsers=ChromeHeadlessCI
-      - run: cd web-ui && npm run e2e -- --protractor-config=e2e/protractor-ci.conf.js
+      # - run: cd web-ui && npm run e2e -- --protractor-config=e2e/protractor-ci.conf.js
```
Comment (AlexITC) [inline .circleci/config.yml:21]: Tests are expected to run in the CI

File: web-ui/src/app/components/nodes/masternode-details/masternode-details.component.html [hunk 39]
```diff
@@ -0,0 +1,52 @@
+<div class="tab-content row ml-0 mr-0">
+	<div class="col-xs-12 text-primary page-title">
+		{{ip ? ip : 'label.masternode' | translate}}
+  </div>
```
Comment (AlexITC) [inline web-ui/src/app/components/nodes/masternode-details/masternode-details.component.html:4]: Please make sure that all the changes have properly formatted code

**Review by AlexITC — CHANGES_REQUESTED** [review]
Overall I took a quick look to this huge change, I'd like to merge it as soon as possible, I'll approve when the PR gets these changes:
- Enable the tests and get them passing in the CI
- Format the code properly, there are some html pieces not aligned.
- Remove unused/commented code
- There are some critical pieces that I want you tor write tests for, like the calculator functions
- Try to avoid using `any` types, and specify types when possible.

Thanks.

File: web-ui/package.json (hunk 29 above)
Comment (midasdev711) [reply web-ui/package.json:8]: That's for staging server. (http://staging.xsnexplorer.io/)
In the staging server, I build the project and run it with node server.
You can check that in server.js.

File: web-ui/src/app/components/calculator/calculator.component.ts (hunk 35 above)
Comment (midasdev711) [reply web-ui/src/app/components/calculator/calculator.component.ts]: Added type.

File: web-ui/src/app/components/calculator/calculator.component.ts (hunk 36 above)
Comment (midasdev711) [reply web-ui/src/app/components/calculator/calculator.component.ts]: Added type.

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts (hunk 31 above)
Comment (midasdev711) [reply web-ui/src/app/components/addresses/address-details/address-details.component.ts]: Added type.
That's to save interval id so that the component can remove interval when it's destroyed.

File: web-ui/package.json (hunk 29 above)
Comment (AlexITC) [reply web-ui/package.json:8]: In my opinion, this is not necessary for the project, I'd suggest keeping this change in your local machine only

Comment (AlexITC) [reply web-ui/package.json:8]: In fact, given that it is staging, why can't you use `ng server` instead?

File: .circleci/config.yml (hunk 38 above)
Comment (midasdev711) [reply .circleci/config.yml:21]: There is no e2e test in the project now. So I have to start from scratch.

Comment (midasdev711) [reply .circleci/config.yml:21]: Can you give me the scope of the test?

Comment (midasdev711) [reply .circleci/config.yml:21]: Which workflow should be tested with protractor?

Comment (AlexITC) [reply .circleci/config.yml:21]: Apologize, you are right that we don't have e2e tests, this is fine.

**Comment by AlexITC** [comment]
Superseded by #329 .
